checker.cpp

```diff
@@ -1,46 +1,87 @@
 #include <iostream>
 #include <assert.h>
 #include <unistd.h>
+#include <map>
 using namespace std;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if(temperature > 102 || temperature < 95) {
-    cout << "Temperature critical!\n";
+string language = "German"; // global variable to set the language
+//Contains Transltons...
+map<string, map<string, string>> translations = {
+    {"English", {
+        {"Temperature critical!", "Temperature critical!"},
+        {"Pulse Rate is out of range!", "Pulse Rate is out of range!"},
+        {"Oxygen Saturation out of range!", "Oxygen Saturation out of range!"}
+    }},
+    {"German", {
+        {"Temperature critical!", "Kritische Temperatur!"},
+        {"Pulse Rate is out of range!", "Pulsfrequenz außerhalb des Bereichs!"},
+        {"Oxygen Saturation out of range!", "Sauerstoffsättigung außerhalb des Bereichs!"}
+    }}
+};
+
+
+void printCriticalMessage(string message) {
+    cout << translations[language][message] << endl;
     for (int i = 0; i < 6; i++)
     {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+        cout << "\r* " << flush;
+        sleep(1);
+        cout << "\r *" << flush;
+        sleep(1);
     }
+}
 
-    return 0;
-  } else if(pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+float convertTemperature(float temperature, string unit) {
```

Comment:
Let the name remove all ambiguity in purpose. The name `convertTemperature` could also mean converting to Kelvin. This detail does matter for a caller of the code, but they will need to look into the implementation to figure out what it is.